src/template.rs

```diff
@@ -0,0 +1,122 @@
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use xml::attribute::OwnedAttribute;
+use xml::reader::XmlEvent;
+use xml::EventReader;
+
+use crate::error::TiledError;
+use crate::{util::*, EmbeddedParseResultType, MapTilesetGid, ObjectData, ResourceCache, Tileset};
+
+/// A template, consisting of an object and a tileset
+///
+/// The path is the path to the template file; it is used to identify them uniquely, and skip parsing the file multiple
+/// times.
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct Template {
+    pub path: PathBuf,
+    pub tileset: Option<Arc<Tileset>>,
+    pub object: Option<ObjectData>,
```

Comment:
The `object` attribute is mandatory, it just wasn't specified in the docs (oops): https://github.com/mapeditor/tiled/commit/8f1162d9cd3e0660d14ac83193c795e825a5a78a